subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -673,6 +834,26 @@ int nrf_cloud_obj_pvt_add(struct nrf_cloud_obj *const obj,
 
 		return nrf_cloud_pvt_data_encode(pvt, obj->json);
 	}
+	case NRF_CLOUD_OBJ_TYPE_COAP_CBOR:
+	{
+		if (!obj->coap_cbor) {
+			return -ENOENT;
+		}
+
+		if (obj->coap_cbor->pvt) {
```

Comment:
instead, check the union type to see if it has already been set:
e.g.: `if (obj->coap_cbor->type != NRF_CLOUD_DATA_TYPE_NONE)`